e3/anod/checkout.py

```diff
@@ -31,6 +31,34 @@ def __init__(self, name, working_dir):
             os.path.join(working_dir, self.name))
         self.metadata_file = self.working_dir + '_checkout.json'
 
+    def update_fingerprint(self, fingerprint, is_prediction=False):
+        """Given a fingerprint amend it with metadata of the current checkout.
+
+        :param fingerprint: initial fingerprint
+        :type fingerprint: Fingerprint
+        :param is_prediction: if True this is an attempt to compute the
+            fingerprint before doing an update. For a checkout the returned
+            value in that case is always the None.
+            See e3.job.walk.compute_fingerprint for more information.
+        :type is_prediction: bool
+        :return: a fingerprint
+        :rtype: None | Fingerprint
+        """
+        if is_prediction:
+            return None
+
+        try:
+            with open(self.metadata_file) as fd:
+                vcs_info = json.load(fd)
+            fingerprint.add(
+                self.name, (vcs_info['url'],
+                            vcs_info['new_commit'],
+                            vcs_info['revision']))
```

Comment:
revision should not be needed, double check for subversion

Reply:
I think we should use `git ls-tree -r HEAD` for Git instead of looking at url and commit id. This is really what we care about for fingerprint.

Reply:
Added for git only